tools/scripts/generate_docker_release_matrix.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+"""Generates a matrix for docker releases through github actions
+
+Will output a condensed version of the matrix. Will include fllowing:
+    * CUDA version short
+    * CUDA full version
+    * CUDNN version short
+    * Image type either runtime or devel
+    * Platform linux/arm64,linux/amd64
+
+"""
+
+import json
+import os
+import sys
+import argparse
+from typing import Dict, List
+
+import generate_binary_build_matrix
+
+DOCKER_IMAGE_TYPES = ["runtime", "devel"]
+
+
+def generate_docker_matrix(channel: str) -> Dict[str, List[Dict[str, str]]]:
+
+    ret: List[Dict[str, str]] = []
+    for cuda in generate_binary_build_matrix.CUDA_ARCHES_DICT[channel]:
+        version = generate_binary_build_matrix.CUDA_CUDDN_VERSIONS[cuda]
+        for image in DOCKER_IMAGE_TYPES:
+            ret.append(
+                {
+                    "cuda": cuda,
+                    "cuda_full_version": version["cuda"],
+                    "cudnn_version": version["cudnn"],
+                    "image_type": image,
+                    "platform": "linux/arm64,linux/amd64",
+                }
+            )
+    return {"include": ret}
+
+
+def main(args) -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--channel",
+        help="Channel to use, default nightly",
+        type=str,
+        choices=["nightly", "test", "release", "all"],
+        default=os.getenv("CHANNEL", "nightly"),
+    )
+    options = parser.parse_args(args)
+
+    build_matrix = generate_docker_matrix(options.channel)
+    print(json.dumps(build_matrix))
+
+if __name__ == "__main__":
+    main(sys.argv[1:])
```

Comment:
This looks weird, I don't think you need to pass `sys.argv` to `argparse` like this, just calling `options = parser.parse_args()` without any arguments works.